js/battledata.js

```diff
@@ -330,8 +330,41 @@ var Tools = {
 				options.hidestrikethrough ? '$1' : '<s>$1</s>');
 		// linking of URIs
 		if (!options.hidelinks) {
-			str = str.replace(/https?\:\/\/[a-z0-9-.]+(?:\:[0-9]+)?(?:\/(?:[^\s]*[^\s?.,])?)?|[a-z0-9.]+\@[a-z0-9.]+\.[a-z0-9]{2,3}|(?:[a-z0-9](?:[a-z0-9-\.]*[a-z0-9])?\.(?:com|org|net|edu|us|jp)(?:\:[0-9]+)?|qmark\.tk)(?:(?:\/(?:[^\s]*[^\s?.,])?)?)\b/ig, function(uri) {
-				if (/[a-z0-9.]+\@[a-z0-9.]+\.[a-z0-9]{2,3}/ig.test(uri)) {
+			var domainRegex = '[a-z0-9\\-]+(?:[.][a-z0-9\\-]+)*';
+			var parenthesisRegex = '[(][^\\s()<>]*[)]';
+			var linkRegex = new RegExp(
+				'\\b' +
+				'(?:' +
+					'(?:' +
+						// When using www. or http://, allow any-length TLD (like .museum)
+						'(?:https?://|www[.])' + domainRegex +
+						'|' + domainRegex + '[.]' +
+							// Allow a common TLD, or any 2-3 letter TLD followed by : or /
+							'(?:com?|org|net|edu|info|us|jp|[a-z]{2,3}(?=[:/]))' +
+					')' +
+					'(?:[:][0-9]+)?' +
+					'\\b' +
+					'(?:' +
+						'/' +
+						'(?:' +
+							'(?:' +
+								'[^\\s()<>]' +
+								'|' + parenthesisRegex +
+							')*' +
+							// URL cannot end with punctation characters
```

Comment:
Also do we want to move this regex definition out of this function? I don't know how good JIT compilers are, but this is a function that's called multiple times per second.